pkg/controller/cluster/server/config.go

```diff
@@ -81,17 +85,47 @@ func serverOptions(cluster *v1beta1.Cluster, token string) string {
 	}
 
 	if cluster.Spec.Mode != agent.VirtualNodeMode {
-		opts = opts + "disable-agent: true\negress-selector-mode: disabled\ndisable:\n- servicelb\n- traefik\n- metrics-server\n- local-storage"
+		opts = opts + "disable-agent: true\negress-selector-mode: disabled\ndisable:\n- servicelb\n- traefik\n- metrics-server\n- local-storage\n"
+	}
+
+	// Adding a check for the version here to workaround issue https://github.com/rancher/k3k/issues/477
+	// in older versions of k3s
+	if requiresNetworkPolicyDisable(cluster) {
+		opts = opts + "disable-network-policy: true\n"
 	}
-	// TODO: Add extra args to the options
 
 	return opts
 }
 
-func configSecretName(clusterName string, init bool) string {
+func ConfigSecretName(clusterName string, init bool) string {
 	if !init {
 		return controller.SafeConcatNameWithPrefix(clusterName, configName)
 	}
 
 	return controller.SafeConcatNameWithPrefix(clusterName, initConfigName)
 }
+
+func requiresNetworkPolicyDisable(cluster *v1beta1.Cluster) bool {
+	imageVersion := "latest"
+
+	if cluster.Spec.Version != "" {
+		imageVersion = cluster.Spec.Version
+	} else if cluster.Status.HostVersion != "" {
+		imageVersion = cluster.Status.HostVersion
+	}
+
+	if imageVersion == "latest" {
+		return false
+	}
+
+	for _, fixedVersion := range k3sNetpolVersions {
+		if semver.MajorMinor(imageVersion) == semver.MajorMinor(fixedVersion) {
+			// if the major and minor match then check if the patch version is less than fixed version
+			version, _ := strings.CutSuffix(imageVersion, "-k3s1")
```

Comment:
I am not familiar with the k3s release naming convention but it looks it happens to have also the `k3s2` metadata ([`v1.29.6+k3s2`](https://github.com/k3s-io/k3s/releases/tag/v1.29.6%2Bk3s2)). In those cases this is not going to work. I would relax the check with a simpler `Cut`:

```suggestion
			version, _, _ := strings.Cut(imageVersion, "-")
```

The `semver` package has also a `Canonical` func, but unfortunately it's not working only with the build metadata. The `-` is used to mark a prerelease instead.
```go
semver.Canonical("v1.33.3-k3s1") // v1.33.3-k3s1
semver.Canonical("v1.33.3+k3s1") // v1.33.3
```